### ISSUE TITLE
Textarea validation

### ISSUE THREAD

**Issue by adrianratajczak** [Dogfalo/materialize#3040]
How to enable validation for textarea? In single input all is working well but on textarea there is no validation. 
Bug?

**Comment by Dogfalo** [comment]
What kind of validation are you expecting?

**Comment by gizm0bill** [comment]
![screen shot 2016-08-18 at 10 10 22 am](https://cloud.githubusercontent.com/assets/189533/17765037/0ca04a62-652c-11e6-9d74-47785b74d92a.png)
![screen shot 2016-08-18 at 10 10 27 am](https://cloud.githubusercontent.com/assets/189533/17765040/0e96ce18-652c-11e6-95b1-96a11bd0a7d5.png)
![screen shot 2016-08-18 at 10 10 36 am](https://cloud.githubusercontent.com/assets/189533/17765045/10c87e8e-652c-11e6-9191-9515120d5718.png)

**Comment by djake** [comment]
Try adding a `validate` class to the textarea. I am only able to get this to trigger the default browser validation when the `required` attribute is present. The `data-error` attribute is not respected, if I manually inject the `valid` or `invalid` css classes, the formatting is all off. Not sure if this is a bug or just needs to be clarified that the docs, or if I'm doing something wrong.

For example:
```
<html>
<head>
  <!-- Compiled and minified CSS -->
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.97.7/css/materialize.min.css">

</head>

<body>
  <!-- Compiled and minified JavaScript -->
	<p>This is a test.</p>
	<form>
<div class="row">
	<div class="input-field col s12">
		<textarea
			class="materialize-textarea validate"
			name="test_textarea"
			placeholder="Write valid stuff"
			required></textarea>
			<label for="test_textarea" data-error="An error has occurred.">Test label</label>
	</div>
</div>
<div class="row">
	<div class="input-field col s12">
		<input type="text" pattern="(only|valid|text)" class="validate"/>
		<label for name="text_test" data-error="An error has occurred.">Text field label</label>
<button type="submit">Submit</button>
</form>
<script type="text/javascript" src="https://code.jquery.com/jquery-2.1.1.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.97.7/js/materialize.min.js"></script>
	</body>
	</html>
```

**Comment by Gorzas** [comment]
I have the same issue. I'm going to check the code to see if I can add this functionality.

**Comment by AlexandreBeauduin** [comment]
Same issue here :)

**Comment by zilions** [comment]
Issue is still present

**Comment by darrylmendonez** [comment]
@djake 's comment worked for me. Add a class="validate" to your textarea tag.

**Comment by pingec** [comment]
@djake's comment only partially fixes it or put another way it introduces another issue. Please take a look at the example. **Without** ```class="validate"``` on ```textarea``` it looks nice but breaks when typing in additional lines.

**With** ```class="validate"``` on ```textarea``` it changes css rules and ```transform: scale(0.8)``` does not apply to it anymore so the font is bigger than other input fields and which breaks the common look principle.

https://codepen.io/pingec/pen/qPqVra

If my conclusion is correct it should be marked as a bug.

**Comment by ashkitten** [comment]
I haven't dug into the source code of Materialize yet, and I don't know if I will, but I threw together a quick fix in the Sass of my website - maybe it'll help someone?
```Sass
textarea.materialize-textarea.validate + label:not(.label-icon).active {
    transform: scale(0.8) translateY(-25px);

    &::after {
        transform: translateY(36px);
    }
}
```
It doesn't quite solve the issue that @pingec mentioned where if the textarea is invalid and is also expanded, then the scaling will cause the error message to appear higher than it should - I assume this is the reason that the developers chose to leave it without scaling in the first place.

**Comment by acburst** [comment]
Fixed in v1-dev aa03660a179d5824847a8179868ba19431605c0b
